subsys/bluetooth/host/hci_core.h

```diff
@@ -468,6 +468,29 @@ uint8_t bt_get_phy(uint8_t hci_phy);
  * @return CTE type (@ref bt_df_cte_type).
  */
 int bt_get_df_cte_type(uint8_t hci_cte_type);
+
+/** Start or restart scanner if needed
+ *
+ * Examples of cases where it may be required to start/restart a scanner:
+ * - When the auto-connection establishement feature is used:
+ *   - When the host sets a connection context for auto-connection establishment.
+ *   - When a connection was established.
+ *     The host may now be able to retry to automatically set up a connection.
+ *   - When a connection was disconnected/lost.
+ *     The host may now be able to retry to automatically set up a connection.
+ *   - When the application stops explicit scanning.
+ *     The host may now be able to retry to automatically set up a connection.
+ *   - The application tries to connect to another device, but fails.
+ *     The host may now be able to retry to automatically set up a connection.
+ * - When the application wants to connect to a device, but we need
+ *   to fallback to host privacy.
+ * - When the application wants to establish a periodic sync to a device
+ *   and the application has not already started scanning.
```

Comment:
For all the cases the stack assumes 1M is used as a primary advertising PHY. I'm not sure if this is intentional, or a bug. 
If we start scanning on Coded PHY if the controller supports it, this may lead to higher power consumption.

Reply:
Yeah since coded was not really supported by bsim, it wasn't tested. We could make a bsim test for this too I guess..

Reply:
Discussed offline. Will add a separate PR to add scanning on Coded PHY when it is likely that the app will use it.

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/pull/72096